### ISSUE TITLE
Builds fails on rawhide because of some wx dependency conflicts

### ISSUE THREAD

**Issue by nickoe** [KiCad/fedora-packaging#29]
It looks like the builds have started to fail a week or two ago because some wxpython dependency conflicts. See log at:

https://copr-be.cloud.fedoraproject.org/results/@kicad/kicad/fedora-rawhide-x86_64/00876075-kicad/builder-live.log

```
Error: 
 Problem: conflicting requests
  - nothing provides python3-sip-api(12)(x86-64) = 12.5 needed by python3-wxpython4-4.0.4-2.fc30.x86_64
(try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
```

Do you have any idea @stevefalco or @aimylios ? Thank you.

**Comment by stevefalco** [comment]
I'll try a local build, and if I get the same error then I'll ask about it on the Fedora Development list.

It looks like KiCad builds correctly for Fedora 30.  Do you agree?

There was a discussion from October 2018 about a similar issue:

https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/thread/IRKUCSICZHCP6JO7NPU4PPV5ELEUPZAV/

But that one was regarding Fedora 30, not Rawhide.

	Steve

**Comment by stevefalco** [comment]
I see the same thing locally, so I just sent the following to the Fedora mailing list:



The KiCad package uses python3-wxpython4 - I'm one of the package maintainers.  KiCad is building correctly on Fedora 30, but my rawhide builds are failing with the following error:

BUILDSTDERR:   - nothing provides python3-sip-api(12)(x86-64) = 12.5 needed by python3-wxpython4-4.0.4-2.fc30.x86_64

It looks like the sip package was upgraded to 12.6 recently (March 21, 2019).  Here is part of the changelog from the sip.spec file:

%changelog
* Thu Mar 21 2019 Rex Dieter <rdieter@fedoraproject.org> - 4.19.15-1
- 4.19.15, - sip-api(12)=12.6

It seems like the python3-wxpython4 package should be rebuilt to pick up that change.

I am new to this part of the process.  How might I request a rebuild of python3-wxpython4, and would that fix the problem or is something additional needed?

	Thanks,
	Steve

**Comment by stevefalco** [comment]
For fun I tried rebuilding python3-wxpython4 to see if I could then use it to build KiCad.  That didn't work - python3-wxpython4 won't build at the moment.  Debian has already seen the issue:

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=924856

It looks like the recent changes to sip have broken python3-wxpython4.  I'll give the Fedora folks a day or two to respond to my previous email.  Failing something helpful there, I'll open a Fedora bug report.

**Comment by aimylios** [comment]
Yes, I already saw that last week, but didn't really care because only rawhide is affected and I expected someone to rebuild python3-wxpython4. Usually the maintainer of a library will trigger a rebuild for all dependent packages in case of a change in API, but apparently that didn't happen in this case.
Also, there seems to be an incompatibility issue with the current version of sip, see https://bugzilla.redhat.com/show_bug.cgi?id=1696302 for reference.

**Comment by nickoe** [comment]
Ok, thank you for the prompt response. I hope it will be solved soon elsewhere then.

**Comment by stevefalco** [comment]
Got a reply on the Fedora list from @swt2c. Here it is for completeness:

The python-wxpython4 package needs to be rebuilt every time that the SIP API changes.  @Rex - can you please initiate a build when you bump the SIP API?

Currently, though, wxPython is FTBFS in Rawhide [1] due to changes in SIP 4.19.14 and 4.19.15.  I've got the SIP 4.19.14 changes addressed in wxPython, but 4.19.15 introduced some bugs in SIP which I'm addressing with SIP upstream.  Hopefully, there should be a new SIP release soon.

Bottom line - the issues are being worked on.  

[1] https://bugzilla.redhat.com/show_bug.cgi?id=1696302

**Comment by stevefalco** [comment]
A new package python3-wxpython4-4.0.4-3.fc31.x86_64.rpm has been built in koji.  It hasn't made it to the repositories yet, but I was able to download it and build KiCad with it.

Thus, in a few days, the nightlies should be back to normal.  We just need to wait for the wxpython package to push out to the mirrors.

**Comment by aimylios** [comment]
The issue seems to be resolved now. Today's rawhide nightly build (with the new python3-wxpython4-4.0.4-3.fc31) was successful [1].

[1] https://copr.fedorainfracloud.org/coprs/g/kicad/kicad/build/883116/

**Comment by nickoe** [comment]
Thank you very much. I will close this issue then ;)

**Comment by stevefalco** [comment]
True, but the F29 build appears to be "stuck".  The other three finished in about 4 hours, but the F29 build is at 15 hours and not done yet.

https://copr-be.cloud.fedoraproject.org/results/@kicad/kicad/fedora-29-x86_64/00883116-kicad/builder-live.log shows that the packages built, but then there was a broken pipe error:

packet_write_wait: Connection to 172.25.80.172 port 22: Broken pipe

So something seems to have gone wrong within Copr.  I don't know if that is due to a disk space issue, or if there is something else going on.

@nickoe - do you want to try killing the F29 build and restarting it?

**Comment by aimylios** [comment]
I think we don't have to worry about this particular problem with the build for F29. The copr system is not very stable, and I see this kind of errors relatively often with my own copr builds (e.g., see today's kicad-nightly [1]). I'm pretty sure that the next build will be fine.

[1] https://copr.fedorainfracloud.org/coprs/aimylios/kicad-nightly/build/883149/

**Comment by stevefalco** [comment]
Thanks @aimylios.  There may be a new issue regarding doxygen.  Some folks are seeing this error:

```
Problem: package doxygen-1:1.8.15-9.fc31.x86_64 requires graphviz, but
none of the providers can be installed
- conflicting requests
- nothing provides urw-base35-fonts needed by graphviz-2.40.1-44.fc31.x86_64

```

The issue was reported on the Fedora development list this morning, but there are no comments as to what might be going wrong just yet.

I just ran a local mock build, and it didn't have any issues.  I've kicked off both a copr build and a koji build to see how they behave, and I'll let you know the results.

**Comment by stevefalco** [comment]
There is an issue with koji right now.  My build failed with:

```
Error: 
 Problem 1: package doxygen-1:1.8.15-9.fc31.aarch64 requires graphviz, but none of the providers can be installed
  - conflicting requests
  - nothing provides urw-base35-fonts needed by graphviz-2.40.1-44.fc31.aarch64
 Problem 2: package asciidoc-8.6.10-0.9.20180605git986f99d.fc30.noarch requires graphviz, but none of the providers can be installed
  - conflicting requests
  - nothing provides urw-base35-fonts needed by graphviz-2.40.1-44.fc31.aarch64
(try to add '--skip-broken' to skip uninstallable packages)
```

Copr seems ok, so the nightlies should be unaffected.


The koji problem is being worked:

```
This was due to a releng PR I merged yesterday to the script that blocks
retired packages. ;(

It blocked a bunch of things that are in a weird state (which we should
totally fix), including that fonts package and all our images. 

I've reverted the PR merge and unblocked all the packages.
Things should return to normal in a bit here.

Sorry for the troubles...

kevin
```
